src/index.ts

```diff
@@ -119,17 +119,67 @@ function makePassesFilter(opts: FilterOpts): (msg: Msg) => boolean {
 class threads {
   private readonly ssb: Record<string, any>;
   private readonly isBlocking: (obj: any, cb: CB<boolean>) => void;
+  private readonly isFollowing: ((obj: any, cb: CB<boolean>) => void) | null;
 
   constructor(ssb: Record<string, any>, _config: any) {
     this.ssb = ssb;
     this.isBlocking = ssb.friends?.isBlocking
       ? ssb.friends.isBlocking
       : IS_BLOCKING_NEVER;
+    this.isFollowing = ssb.friends?.isFollowing;
     this.ssb.db.registerIndex(HashtagsPlugin);
   }
 
   //#region PRIVATE
 
+  private maybeOnlyKeepFollowingMessages =
+    (enabled: boolean) => (source: any) =>
+      pull(
+        source,
+        pull.asyncMap((msg: Msg, cb: CB<Msg | null>) => {
+          if (!enabled) {
```

Comment:
We can move this check up, before `pull()` on line 137 is called. It could be `enabled ? pull.through() : pull( source, ...`

Reply:
ah `pull.through` was what I was looking for. would it make sense to avoid the `enabled` arg here altogether and handle this one level up in the public methods?